### ISSUE TITLE
libproj 9.4.0 causes test failures due to updated proj-db values

### ISSUE THREAD

**Issue by urschrei** [georust/docker-images#28]
New images build successfully, but can't be tested using the existing proj crate tests. Example:

```
thread 'proj::test::test_array_convert' panicked at src/proj.rs:1492:9:
assert_relative_eq!(v[1].y(), 1141293.7960220438)

    left  = 1141293.7960220457
    right = 1141293.7960220438
```

WDYT @michaelkirk: should I publish them and we can then fix the tests on the Rust side?

**Comment by michaelkirk** [comment]
SGTM

**Comment by michaelkirk** [comment]
I'm assuming the order of operations is something like:

- publish the new containers with a tag like `georust/proj-ci:proj-9.4.0-rust-1.70`
- update https://github.com/georust/proj/pull/196 to use the new containers
- update the tests in https://github.com/georust/proj/pull/196 to use the slightly new expected values
- merge #196

- at some point in this process, push up the changes to this repository to include the latest proj builds (and rust versions if you're feeling fun).

LMK if you want to hand off the container building process - I have a decently robust x86 linux server that could pump out the new containers decently fast.

**Comment by urschrei** [comment]
I'm just finishing up on pushing up the new containers. I hadn't decided whether to fix up the tests in #196 or as part of a new PR to update `proj` – it feels vaguely wrong to pollute one crate with changes from another, but I think the larger issue is that we should probably rethink our proj-sys testing strategy in order not to depend on tests in proj.

**Comment by urschrei** [comment]
https://github.com/georust/docker-images/pull/29

**Comment by michaelkirk** [comment]
Am I misunderstanding? You're saying tests in the `proj-sys` crate depend on the `proj` crate?  I can't find where that's happening.

My understanding is that updating to this libproj will cause our existing tests to fail because the output of libproj is slightly different.

Since we are updating libproj in https://github.com/georust/proj/pull/196 to produce the new values, to me it seems completely uncontroversial that we'd also update the expected values in that same pr. Am I missing something?

**Comment by urschrei** [comment]
Sorry, not the tests in the crate: CI uses the same test suite for `proj-sys` and `proj`, which means that you can't update `proj-sys` contingent on it passing CI without updating `proj` in the same PR if the libproj update causes tests to fail due to differing values.

Of course after writing that, I see that CI is passing on all Linux hosts (my local tests on macOS are failing), so maybe this is all NBD.

**Comment by michaelkirk** [comment]
For some reason it never occurred to me that it was anything but natural to update proj and proj-sys together in a single PR. LMK if it gets weird - I'd be interested in better understanding the challenge there.

In any case, glad tests are passing now!

I recall too that I've seen slightly different results on aarch64. Kind of spooky, but it's at quite high precision, so it never gave me much pause.

**Comment by urschrei** [comment]
Only current weirdness is here: https://github.com/georust/proj/actions/runs/8510902645/job/23309460967
